.template/variants/web/app/javascript/build.js

```diff
@@ -2,12 +2,18 @@ const watch = process.argv.slice(2).includes('--watch');
 
 require('esbuild')
   .context({
-    entryPoints: ['app/javascript/application.js'],
+    entryPoints: [
+      { in: 'app/javascript/application.js', out: 'application.js' },
+      // { in: 'engines/[engine_name]/app/javascript/[engine_name]/application.js', out: '[engine_name]/application.js' },
+    ],
     inject: ['app/javascript/global.js'],
     bundle: true,
     sourcemap: true,
     minify: !watch,
     outdir: 'app/assets/builds',
+    alias: {
+      'Core': './app/javascript/',
```

Comment:
I also take this chance to add the Alias here, so the future developer no need to research for it, let me know if we would like to remove it from the template 👌 

The alias will allow us to refer to the core js path from the engine JS file with the `Core/components/....` like the image bellow 

<img width="956" alt="image" src="https://github.com/nimblehq/rails-templates/assets/11751745/42549eab-848c-4515-88c4-f253172c873e">

Reply:
Should we rename it `MainApp` or `Root`?

`Core` means "central", but in the case of a simple presentation layer App that relies on `gems` or `engines` for the core, the `main_app` might not be the `core` of the system 💭

Reply:
I am not sure about the usage of [`alias`](https://esbuild.github.io/api/#alias) here.  According to the doc:

> This feature lets you substitute one package for another when bundling. One use case for this feature is replacing a node-only package with a browser-friendly package in third-party code that you don't control.

I think what we're trying to achieve here belongs to the `path` of [`jsconfig.json`](https://stackoverflow.com/questions/68675994/what-is-jsconfig-json) inside the Rails project and [esbuild supports jsconfig.json path as well](https://github.com/evanw/esbuild/issues/38).

An example of `jsconfig.json` for this case could be:

```json
{
  "compilerOptions": {
    "baseUrl": "./app/javascript",
    "paths": {
      "Core/helpers/*": ["helpers/*"]
    }
  }
}
```

Then we could import the helpers exactly how we're doing right now!

Reply:
Nice catch Rajib 🙇

Reply:
I'm not aware about that @tyrro, let me check that

Reply:
Hi @tyrro I tried that and seems it doesn't work FYI

![image](https://github.com/nimblehq/rails-templates/assets/11751745/0b7b1869-c810-402c-8034-d872c6d4d83e)

I will remove the changes in this PR FYI

Reply:
@tyrro I removed the config in this [commit](https://github.com/nimblehq/rails-templates/pull/427/commits/bec259c6835c8f0f09a53a547e1919724a32698c), cc @malparty if you think this is needed for the template, please create the new issue for that 👌

Reply:
Not a must-have, it is quite an opinionated topic from what I remember ^^